### PR TITLE
fix(data-table): use padding instead of fixed cell heights

### PR DESCRIPTION
**Pull request by mediremi**
Closes https://jira.dhis2.org/browse/LIBS-236

This PR replaces the use of fixed height table cells with a less brittle padding-based approach.

The primary benefit of this change is the elimination of overflow-related issues:

| Before | After |
| --------- | ------ |
| image | image |
| image  | image |

### CODE REVIEW

**Review by Mohammer5 — COMMENTED** [review]
I'm wondering if the text in the header cells should be top-aligned. Question for @cooper-joe

**Comment by cooper-joe** [comment]
> I'm wondering if the text in the header cells should be top-aligned.

Yes, I think so. The content (text, optional icons for sorting and filtering) in the header cells should be top-aligned.

**Review by HendrikThePendric — CHANGES_REQUESTED** [review]
LGTM. But I think two points still need to be addressed:
- [ ] Add a story to demo the text overflow
- [ ] Align top instead of align middle

**Comment by HendrikThePendric** [comment]
I worked on getting things to align at the top and started to notice another minor issue: the buttons in the sort/filter icon had a visual regression when we moved from height to padding. This turned out to be surprisingly hard to fix without breaking something else. When I discussed the issue with @mediremi we also cam to realise that having full-height buttons gets very weird when having table headers with text that spans multiple lines. @mediremi suggested that probably a square filter button looks just as nice, perhaps nicer, and I think I agree. So this is what I implemented.

@cooper-joe could you have a look at the sort/filter icon-buttons in the current implementation? It's definitely different from the design specs, but I quite like it. We will hold off merging this until we have your seal of approval.

**Comment by HendrikThePendric** [comment]
@cooper-joe for some reason the PR preview doesn't seem to be updating, I was still looking at the old storybook. If you try the link from the PR preview, please verify you are seeing a story called "Long Cell Content". If not, you are looking at stale code, and I'm afraid you will need to check out the branch and run the storybook yourself.

**Review by HendrikThePendric — APPROVED** [review]
Personally, I am happy with this now.

File: components/table/src/data-table/data-table-column-header/data-table-column-header.styles.js
Comment (HendrikThePendric) [reply components/table/src/data-table/data-table-column-header/data-table-column-header.styles.js:24]: Good catch. I now have 6px form normal cells and 12px for the large cells. Resulting in heights of 37 and 49px respectively (incl. border). This is the same as on master.

**Pull request merged**
